packages/api-server/api_server/repositories/tasks.py

```diff
@@ -119,6 +97,43 @@ async def save_task_state(self, task_state: TaskState) -> None:
             self.logger.error(
                 f"Failed to save task state of id [{task_state.booking.id}] [{e}]"
             )
+            return
+
+        if not created:
+            return
```

Comment:
iiuc, task labels is created when we create a task and never changes in task state updates, could we move label processing to when we dispatch tasks instead?

Reply:
no, when we dispatch a task we will not know the task ID to save these labels to, only when the dispatcher replies with the first task state for saving, will we be able to save the labels

Reply:
Added more comments https://github.com/open-rmf/rmf-web/pull/912/commits/7c2d8efa27ad3edacc438dec24e756bda3c2baa9